src/expression_evaluator/literal_evaluator.cpp

```diff
@@ -19,7 +19,11 @@ bool LiteralExpressionEvaluator::select(
 
 void LiteralExpressionEvaluator::resolveResultVector(
     const processor::ResultSet& /*resultSet*/, MemoryManager* memoryManager) {
-    resultVector = std::make_shared<ValueVector>(*value->getDataType(), memoryManager);
+    if (value->getDataType()->getLogicalTypeID() == LogicalTypeID::ANY) {
```

Comment:
This is not the ideal place to resolve type. All type for an expression/function should be resolved in binder/expression_binder.

Most likely what need to happen is, after we bind `NULL IS NULL`, we need an additional check over the expression to tree to make sure all child has a valid type (i.e. not LogicalTypeID::ANY). If they don't have one, we assign an arbitrary one.

Reply:
I saw duckdb has a NULL datatype. Should we add it as welll to solve such case?

Reply:
I've moved the type resolution to the expression binder.

I also tried implementing a NULL datatype, but it seems that I would need to go through a far larger portion of the codebase to get it working. To do that, I'd need to know how a NULL type would differ from an ANY type, other than maybe its ability to be resolved to a fixed size?